### PR TITLE
chore(i18n): Adds support for all languages on Transifex

### LINKED ISSUE #7154: Implement a test for correct naming (county codes used) of language files
This is a continuation / separate summary of the discussion started at #7146.

Problem: currently there is no definite (Elgg) standard of how to deal with country codes with more than 2 characters (e.g. pt_BR or ja_JP). Also there is no test to make sure than any language file pulled from Transifex is named correctly and to test if there might be language files included for basically the same language but using short and long country code respectively (e.g. pt/pt_PT or ja/ja_JP).

The necessity to decide on a (new) standard / list of supported languages has already been pointed out in #6856 (still open) as ISO 639-1 is outdated. A test of correct naming seems an additional necessity now as language files are already pulled in from Transifex.

For implementing a test it seems necessary to:

* decide on a standard of supported languages / country codes. While the language codes used by Transifex are no official standard it seems a likely choice right now with relying on Transifex.

* adding the list of country codes somewhere to be useable for the test (currently the country code language strings are included in the main language file but the list is limited to two-character codes. I don't know how this list could be used for a test, so a separate updated list might be necessary or the language strings should be saved separately. Indenpendently of a test it seems necessary to update the county code language strings anyway to also include the new languages).

* decide on a language file naming convention. First of all the naming convention has to be based on the country codes standard in use. But it's also necessary to decide on how upper/lower case issues are to be dealt with (currently upper case letters in language file names are converted to lower case). And it's to be decided on how non-letter characters are handled (currently there's a differentiation between "_", e.g. pt_BR.php, and "-"). In connection with deciding on a naming convention it might also be worth taking #6712 into account that arised with Elgg 1.9.

* implement the test itself. Well, I've no idea how to even start with that as I've never done that...

### LINKED ISSUE #7540: Transifex breaks sprintf format/dollar signs in translations
Change lang file name of portuguese files to "pt_br" (currently this as pt_BR). for this reason the translation is not working.

### LINKED ISSUE #8683: Allow more locale codes
There's a huge list of locales in iOS: http://stackoverflow.com/a/28357857/3779

From Transifex we're receiving translations in Serbian (Latin), Chinese (Simplified Han), and Romanian (Romania). While it would be nice to stick to two-char language codes, this is totally unrealistic, and the Elgg core team is in no way qualified to say which codes should be considered too specific.

I think we should pull all those codes ([that are supported by Transifex](https://www.transifex.com/languages/)) in (in a [normalized form](https://github.com/Elgg/Elgg/pull/8681/files#diff-2d105b58e1b35c054f963a1dd752c614R577)) and the community members that speak those languages should sort out which locales they want to put effort into.

What's the downside here?

### PR DESCRIPTION
**Pull request by mrclay**
Adds languages from Transifex; updates release script to normalize language filenames and remove invalid files.

Fixes #7540
Fixes #7154
Fixes #8683
- [ ] add_translation() uses strtolower(). Figure out if this is a problem.
- [ ] gain consensus on pulling in all these codes. Notably there's a client-side cost.

### CODE REVIEW

**Comment by mrclay** [comment]
This is very close! PTAL. Considering trying to normalize codes from Transifex to `[a-z_]`

**Comment by mrclay** [comment]
This now normalizes files to the ones listed in Translator.

**Comment by ewinslow** [comment]
LGTM

File: .scripts/release.php [hunk 1]
```diff
@@ -45,10 +47,19 @@ function run_commands($commands) {
...
-
 // Update translations
 run_commands(array(
 	"tx pull -a --minimum-perc=100",
+));
+
+// Clean translations
+$cleaner = new Elgg\I18n\ReleaseCleaner();
+$cleaner->cleanInstallation(dirname(__DIR__));
+foreach ($cleaner->log as $msg) {
```
Comment (ewinslow) [inline .scripts/release.php]: didnt want to inject a logger for this?

Comment (mrclay) [inline .scripts/release.php]: Our logger has too many dependencies.

**Comment by mrclay** [comment]
Ultimately do we want to accept these new lang codes?

**Comment by iionly** [comment]
I noticed that you added "ro_ro", "sr_latn" and "zh_hans" to the array of allowed languages. Is this really the common intention?

For example I don't know if "ro_ro" isn't just an alias for "ro" and possibly the same with the other longer codes that have a two-character code already. If the longer codes would just be alternatives for the short codes I don't see any necessity to add these new codes. For exampe "de_de" is just a longer code for "de" and there wouldn't be any necessity to add it ever.

And in case these get added it would also be necessary to add language strings for these new languages itself to allow for providing the text to be shown in the language selection (or the raw string would show) and to allow for providing translations, too.

**Comment by mrclay** [comment]
Please see #8683

**Comment by ewinslow** [comment]
I'd consider this a feature. Support for hundreds more language codes is great!

**Comment by mrclay** [comment]
I'd like to get more input on this.

File: .scripts/transifex/parse.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Open https://www.transifex.com/languages/
+// Open console
+// Run this:
+
+i18n_codes = (function(){
+	var codes = {};
+	$("#txlanguage-list li").each(function () {
```
Comment (ewinslow) [inline .scripts/transifex/parse.js]: This kind of code makes me nervous because it seems brittle. This isn't run on every release is it?

Comment (mrclay) [inline .scripts/transifex/parse.js]: No. Just a one-time scrape script

**Comment by juho-jaakkola** [comment]
Closing because a different approach was chosen: #8954

**Pull request closed without merging**